app/org/sagebionetworks/bridge/models/CriteriaUtils.java

```diff
@@ -71,6 +72,22 @@ public static void validate(Criteria criteria, Set<String> dataGroups, Errors er
         }
         validateDataGroups(errors, dataGroups, criteria.getAllOfGroups(), "allOfGroups");
         validateDataGroups(errors, dataGroups, criteria.getNoneOfGroups(), "noneOfGroups");
+        validateDataGroupNotRequiredAndProhibited(criteria, errors);
+    }
+
+    /**
+     * Can't logically have a data group that is both required and prohibited, so check for this.
+     * @param criteria
+     * @param errors
+     */
+    private static void validateDataGroupNotRequiredAndProhibited(Criteria criteria, Errors errors) {
+        if (criteria.getAllOfGroups() != null && criteria.getNoneOfGroups() != null) {
+            Set<String> intersection = Sets.newHashSet(criteria.getAllOfGroups());
+            intersection.retainAll(criteria.getNoneOfGroups());
```

Comment:
Minor: Guava's Sets.intersection() does exactly what you want it to do https://code.google.com/p/guava-libraries/wiki/CollectionUtilitiesExplained#Sets